addons/stock/models/stock_picking.py

```diff
@@ -641,6 +643,125 @@ def action_cancel(self):
         self.write({'is_locked': True})
         return True
 
+    def action_return(self):
+        self.ensure_one()
+        new_picking, return_picking_type = self._action_return()
```

Comment:
are you sure you need to create and redirect to the record? why not redirectingto a "new" record (not yet saved in database) but with proper default values?